cmd/umount.go

```diff
@@ -91,5 +96,89 @@ func umount(ctx *cli.Context) error {
 	setup(ctx, 1)
 	mp := ctx.Args().Get(0)
 	force := ctx.Bool("force")
+	if !force {
+		raw, err := os.ReadFile(path.Join(mp, ".config"))
+		if err != nil {
+			if os.IsNotExist(err) {
+				return fmt.Errorf("not a JuiceFS mount point")
+			}
+			return errors.Wrap(err, "failed to read config")
+		}
+
+		var conf vfs.Config
+		if err = json.Unmarshal(raw, &conf); err != nil {
+			return errors.Wrap(err, "failed to parse config")
+		}
+		if conf.Chunk.Writeback {
+			stagingDir := path.Join(conf.Chunk.CacheDir, "rawstaging")
+			if err := waitWritebackComplete(stagingDir); err != nil {
+				return err
+			}
+			defer func() {
+				size, _ := fileSizeInDir(stagingDir)
+				clearLastLine()
+				if size == 0 {
+					fmt.Println("\rAll staging chunks are written back")
+				} else {
+					fmt.Printf("\r%s staging chunks are not written back\n", humanize.IBytes(size))
+				}
+			}()
+		}
+	}
 	return doUmount(mp, force)
 }
+
+func waitWritebackComplete(stagingDir string) error {
+	lastLeft := uint64(0)
+	for {
+		_, err := os.Stat(stagingDir)
+		if err != nil {
+			if os.IsNotExist(err) {
+				return nil
+			}
+			return errors.Wrap(err, "failed to read staging directory")
+		}
+		start := time.Now()
+		size, err := fileSizeInDir(stagingDir)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			return errors.Wrap(err, "failed to read staging directory")
+		}
+		if lastLeft == 0 {
+			lastLeft = size
+		}
+
+		if size == 0 {
```

Comment:
it's better to double check, wait until both lastLeft and size are 0

Reply:
Done